src/qtt/instrument_drivers/gates.py

```diff
@@ -246,9 +246,22 @@ def get_boundaries(self):
             boundaries[gate] = param.vals.valid_values
         return boundaries
 
+    def restrict_boundaries(self, gate_boundaries):
+        """ Restrict boundaries values that can be set on the gates.
+
+        Args:
+            gate_boundaries (dict): a range of allowed values per parameter.
```

Comment:
gate_boundaries is not the range that will be allowed. It must fit within the current boundary of that gate. This docstring fits the method set_boundaries, not restrict_boundaries